src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,139 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private Object[] objects;
```

Comment:
The name 'objects' is not very informative. Consider renaming it to something more descriptive, like 'elements' or 'items', to clearly represent its purpose.